src/actions/proxy.ts

```diff
@@ -0,0 +1,161 @@
+
+  return class extends Base implements IProxyActions {
+
+    getAccessTokenOwner = async (tokenId: string): Promise<BN> => {
+      const res : { 0: BN } = await executeAndRetry(this.contracts['PROXY_REGISTRY'].ownerOf, [tokenId]);
+      return res[0];
+    }
+
+    getNFTOwner = async (tokenId: string): Promise<BN> => {
+      const res : { 0: BN } = await executeAndRetry(this.contracts['COLLATERAL_NFT'].ownerOf, [tokenId]);
+      return res[0];
+    }
+
+    transferNFT = async (from: string, to: string, tokenId: string) => {
+      const txHash = await executeAndRetry(this.contracts['COLLATERAL_NFT'].transferFrom, [from, to, tokenId, this.ethConfig]);
+      console.log(`[NFT Approve] txHash: ${txHash}`);
+      return waitAndReturnEvents(this.eth, txHash, this.contracts['COLLATERAL_NFT'].abi, this.transactionTimeout);
+    }
+
+    buildProxy = async (owner: string) => {
+      const txHash = await executeAndRetry(this.contracts['PROXY_REGISTRY'].build, [owner, this.ethConfig]);
+      console.log(`[Proxy created] txHash: ${txHash}`);
+      const response: any = await waitAndReturnEvents(this.eth, txHash, this.contracts['PROXY_REGISTRY'].abi, this.transactionTimeout);
+      return response.events[0].data[2].toString();
+    }
+
+    getProxy = async (accessTokenId: string) => {
+      const res = await executeAndRetry(this.contracts['PROXY_REGISTRY'].proxies,  [accessTokenId, this.ethConfig]);
+      return res[0];
+    }
+
+    getProxyAccessToken = async (proxyAddr: string) => {
+      const proxy: any = this.eth.contract(this.contractAbis['PROXY']).at(proxyAddr);
+      const res = await executeAndRetry(proxy.accessToken, [this.ethConfig]);
+      return res[0].toNumber();
+    }
+
+    getProxyFromLoan = async (loanId: string) => {
+      let address;
+      let accessTokenId;
+      try {
+        const res = await executeAndRetry(this.contracts['TITLE'].ownerOf, [loanId]);
+        address = res[0];
+      } catch (e) {
+        address = ZERO_ADDRESS;
+      }
+      if (address !== ZERO_ADDRESS) {
+        try {
+          accessTokenId = await this.getProxyAccessToken(address);
+        } catch (e) {
+          accessTokenId = null;
+        }
+      }
+      if (accessTokenId) {
+        return await this.getAccessTokenOwner(accessTokenId);
+      }
+    }
+
+    proxyCount = async (): Promise<BN> => {
+      const res: { 0: BN }  = await executeAndRetry(this.contracts['PROXY_REGISTRY'].count, []);
+      return res[0];
+    }
+
+    checkProxyExistence = async (borrowerAddr: string) => {
+      let proxy;
+      const count = (await this.proxyCount()).toNumber();
+      for (let i = 0; i < count; i += 1) {
+        const ownerBN = await this.getAccessTokenOwner(i.toString())
+        if (ownerBN && ethers.utils.getAddress(ownerBN.toString()) === ethers.utils.getAddress(borrowerAddr)) {
+          const proxyAddress = await this.getProxy(i.toString());
+          proxy = proxyAddress;
```

Comment:
break; or directly return the proxy, because you found it already. No need to continue the loop :)